src/lookup.py

```diff
@@ -687,162 +700,362 @@ def matches(self, entitytype=None, variant=None, subtype=None):
+            query = f"entity[@id='{adjustedId}'][@variant='{variant}']"
+
+            validMissingSubtype = False
+
+            if entities2Root:
+                entityXML = entities2Root.find(query + f"[@subtype='{subtype}']")
+
+                if entityXML is None:
+                    entityXML = entities2Root.find(query)
+                    validMissingSubtype = entityXML is not None
+
+                if entityXML is None:
+                    printf(
+                        "Loading invalid entity (no entry in entities2 xml): "
+                        + str(node.attrib)
+                    )
+                    node.set("Invalid", "1")
+                else:
+                    foundName = entityXML.get("name")
+                    givenName = node.get("Name")
+                    foundNameClean, givenNameClean = list(
+                        map(
+                            lambda s: self.ENTITY_CLEANUP_REGEX.sub("", s).lower(),
+                            (foundName, givenName),
                         )
-                        entity.set("Invalid", "1")
-                    else:
-                        foundName = entityXML.get("name")
-                        givenName = entity.get("Name")
-                        foundNameClean, givenNameClean = list(
-                            map(
-                                lambda s: cleanUp.sub("", s).lower(),
-                                (foundName, givenName),
+                    )
+                    if not (
+                        foundNameClean == givenNameClean
+                        or (
+                            validMissingSubtype
+                            and (
+                                foundNameClean in givenNameClean
+                                or givenNameClean in foundNameClean
                             )
                         )
-                        if not (
-                            foundNameClean == givenNameClean
-                            or (
-                                validMissingSubtype
-                                and (
-                                    foundNameClean in givenNameClean
-                                    or givenNameClean in foundNameClean
-                                )
-                            )
-                        ):
-                            printf(
-                                "Loading entity, found name mismatch! In entities2: ",
-                                foundName,
-                                "; In BR: ",
-                                givenName,
-                            )
+                    ):
+                        printf(
+                            "Loading entity, found name mismatch! In entities2: ",
+                            foundName,
+                            "; In BR: ",
+                            givenName,
+                        )
+
+        overwrite = node.get("Overwrite") == "1"
```

Comment:
what's the intended use-case for this? Definitely a NicheFeatures candidate

Reply:
This is used to specify when an entity is being intentionally overwritten. Mostly added it to suppress errors in BR since the Lil' Battery and Grab Bag are later overwritten by the Random Battery and Random Grab Bag entities in Repentance. Could alternatively just load those in a separate file similar to secret room types?

Reply:
Ah yes, that would be a reasonable use-case. Speaking of which, the card subtypes change in the various DLCs as well, which idk if you observed (haven't checked xmls yet)